### ISSUE TITLE
DRKG predicate mappings

### ISSUE THREAD

**Issue by schatzkara** [biolink/biolink-model#662]
As part of a research project, I am using DRKG (The Drug Repurposing Knowledge Graph)  and am hoping to map it to biolink so that I can do some analysis comparing DRKG and ROBOKOP. 

Would it be possible to add predicate mappings for the relations in DRKG that are not already mapped? They are listed below.

- DRUGBANK::carrier::Compound:Gene
- DRUGBANK::ddi-interactor-in::Compound:Compound
- DRUGBANK::enzyme::Compound:Gene
- DRUGBANK::target::Compound:Gene
- DRUGBANK::x-atc::Compound:Atc
- DRUGBANK::treats::Compound:Disease
- GNBR::A+::Compound:Gene
- GNBR::A-::Compound:Gene
- GNBR::B::Compound:Gene
- GNBR::B::Gene:Gene
- GNBR::C::Compound:Disease
- GNBR::D::Gene:Disease
- GNBR::E+::Compound:Gene
- GNBR::E+::Gene:Gene
- GNBR::E-::Compound:Gene
- GNBR::E::Compound:Gene
- GNBR::E::Gene:Gene
- GNBR::G::Gene:Disease
- GNBR::H::Gene:Gene
- GNBR::I::Gene:Gene
- GNBR::J::Compound:Disease
- GNBR::J::Gene:Disease
- GNBR::K::Compound:Gene
- GNBR::L::Gene:Disease
- GNBR::Md::Gene:Disease
- GNBR::Mp::Compound:Disease
- GNBR::N::Compound:Gene
- GNBR::O::Compound:Gene
- GNBR::Pa::Compound:Disease
- GNBR::Pr::Compound:Disease
- GNBR::Q::Gene:Gene
- GNBR::Rg::Gene:Gene
- GNBR::Sa::Compound:Disease
- GNBR::T::Compound:Disease
- GNBR::Te::Gene:Disease
- GNBR::U::Gene:Disease
- GNBR::Ud::Gene:Disease
- GNBR::V+::Gene:Gene
- GNBR::W::Gene:Gene
- GNBR::X::Gene:Disease
- GNBR::Y::Gene:Disease
- GNBR::Z::Compound:Gene
- GNBR::in_tax::Gene:Tax
- Hetionet::AdG::Anatomy:Gene
- Hetionet::AeG::Anatomy:Gene
- Hetionet::AuG::Anatomy:Gene
- Hetionet::CbG::Compound:Gene
- Hetionet::CcSE::Compound:Side Effect
- Hetionet::CdG::Compound:Gene
- Hetionet::CpD::Compound:Disease
- Hetionet::CrC::Compound:Compound
- Hetionet::CtD::Compound:Disease
- Hetionet::CuG::Compound:Gene
- Hetionet::DaG::Disease:Gene
- Hetionet::DdG::Disease:Gene
- Hetionet::DlA::Disease:Anatomy
- Hetionet::DpS::Disease:Symptom
- Hetionet::DrD::Disease:Disease
- Hetionet::DuG::Disease:Gene
- Hetionet::GcG::Gene:Gene
- Hetionet::GiG::Gene:Gene
- Hetionet::GpBP::Gene:Biological Process
- Hetionet::GpCC::Gene:Cellular Component
- Hetionet::GpMF::Gene:Molecular Function
- Hetionet::GpPW::Gene:Pathway
- Hetionet::Gr>G::Gene:Gene
- Hetionet::PCiC::Pharmacologic   Class:Compound
- INTACT::ADP RIBOSYLATION   REACTION::Gene:Gene
- INTACT::ASSOCIATION::Compound:Gene
- INTACT::ASSOCIATION::Gene:Gene
- INTACT::CLEAVAGE REACTION::Gene:Gene
- INTACT::COLOCALIZATION::Gene:Gene
- INTACT::DEPHOSPHORYLATION   REACTION::Gene:Gene
- INTACT::DIRECT INTERACTION::Compound:Gene
- INTACT::DIRECT INTERACTION::Gene:Gene
- INTACT::PHOSPHORYLATION   REACTION::Gene:Gene
- INTACT::PHYSICAL ASSOCIATION::Compound:Gene
- INTACT::PHYSICAL ASSOCIATION::Gene:Gene
- INTACT::PROTEIN CLEAVAGE::Gene:Gene
- INTACT::UBIQUITINATION REACTION::Gene:Gene
- STRING::ACTIVATION::Gene:Gene
- STRING::BINDING::Gene:Gene
- STRING::CATALYSIS::Gene:Gene
- STRING::EXPRESSION::Gene:Gene
- STRING::INHIBITION::Gene:Gene
- STRING::OTHER::Gene:Gene
- STRING::PTMOD::Gene:Gene
- STRING::REACTION::Gene:Gene
- bioarx::Coronavirus_ass_host_gene::Disease:Gene
- bioarx::Covid2_acc_host_gene::Disease:Gene
- bioarx::DrugHumGen:Compound:Gene
- bioarx::DrugVirGen:Compound:Gene
- bioarx::HumGenHumGen:Gene:Gene
- bioarx::VirGenHumGen:Gene:Gene

@balhoff

**Comment by cmungall** [comment]
Can you help us understand the format?

Are these lists of actual predicates

I am not sure how to intrepret:

GNBR::E+::Compound:Gene
GNBR::E+::Gene:Gene
GNBR::E-::Compound:Gene
GNBR::E::Compound:Gene

**Comment by schatzkara** [comment]
Yes, of course. Sorry I should have added this to the initial issue.
The format is: source::predicate::connected_entity_type1:connected_entity_type2

Specifically, the ones that you mentioned come from GNBR which uses the following key for the predicate types.
[GNBR Predicate Key](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6937428/figure/F1/?report=objectonly)

The predicates from Hetionet are similarly confusing and are defined by the following key:
[Hetionet Predicate Key](https://github.com/hetio/hetionet/blob/master/describe/edges/metaedges.tsv)

**Comment by sierra-moxon** [comment]
@schatzkara - I imagine we'd use full predicate names like 'agonism' vs. 'A+' (we could potentially add things like 'A+' as an alias? - would this work?)

**Comment by schatzkara** [comment]
Yes, full names would definitely be fine as opposed to the far less readable acronyms :)

**Comment by mbrush** [comment]
Note that some of these predicates fall into the 'interaction mechanism' category, which the Chemical WG is working to harmonize.  so work on these should be seeded to / coordinated with them.  

Their working s/s is [HERE](https://docs.google.com/spreadsheets/d/1VI2s5YVDwOHiQv7hBWkl12d86yWvm-z_pDJFnDwsoiY/edit#gid=385022388). Note that GNBR is not one of the sources they targeted for harmonization - so we should be sure to add the GNBR interaction mechanism terms to the harmonization s/s . I added a stub sheet to the doc with a link to the GNBR predicate list.

**Comment by sierra-moxon** [comment]
Thanks @mbrush!  I'll note the work we did in the datamodeling helpdesk this morning on drug->gene interaction on the GNBR tab of that spreadsheet. :)

**Comment by sierra-moxon** [comment]
tagging @ericawood as she had a ticket that was very similar here: #679

**Comment by nlharris** [comment]
Can this be closed?

**Comment by sierra-moxon** [comment]
@schatzkar, I believe much of this case is handled by [this PR](https://github.com/biolink/biolink-model/pull/844/files) (refactoring the chemical hierarchy).  Would your team have a chance to review that PR with regard to this ticket?  I've attached this ticket there, so that when the PR is merged, this ticket will be resolved.

**Comment by schatzkara** [comment]
> @schatzkar, I believe much of this case is handled by [this PR](https://github.com/biolink/biolink-model/pull/844/files) (refactoring the chemical hierarchy). Would your team have a chance to review that PR with regard to this ticket? I've attached this ticket there, so that when the PR is merged, this ticket will be resolved.

@sierra-moxon It looks like some of the DRUGBANK predicates and several of the Hetionet predicates are covered there, but some others are not. I did not see the GNBR, INTACT, STRING, or bioarx predicates there. I can give a more specific list if that's what you're looking for. Maybe the rest do not fit into the ontology?
